dojo/unittests/tools/test_sarif_parser.py

```diff
@@ -297,25 +326,43 @@ def test_flawfinder(self):
+**Rule name:** random/setstate
+**Rule short description:** This function is not sufficiently random for security-related functions such as key and nonce creation (CWE-327).
+'''
+            self.assertEqual(description, finding.description)
             self.assertEqual("src/tree/param.cc", finding.file_path)
             self.assertEqual(29, finding.line)
             self.assertEqual(327, finding.cwe)
             self.assertEqual("FF1048", finding.vuln_id_from_tool)
+            self.assertEqual('https://cwe.mitre.org/data/definitions/327.html', finding.references)
         with self.subTest(i=20):
             finding = findings[20]
             self.assertEqual("buffer/memcpy:Does not check for buffer overflows when copying to destination (CWE-120).", finding.title)
             self.assertEqual("Info", finding.severity)
-            self.assertEqual("Does not check for buffer overflows when copying to destination (CWE-120).", finding.description)
+            description = '''**Result message:** buffer/memcpy:Does not check for buffer overflows when copying to destination (CWE-120).
+**Snippet:**     std::memcpy(dptr, dmlc::BeginPtr(buffer_) + buffer_ptr_, size);
+**Rule name:** buffer/memcpy
+**Rule short description:** Does not check for buffer overflows when copying to destination (CWE-120).
+'''
+            self.assertEqual(description, finding.description)
             self.assertEqual("src/common/io.cc", finding.file_path)
             self.assertEqual(31, finding.line)
             self.assertEqual(120, finding.cwe)
             self.assertEqual("FF1004", finding.vuln_id_from_tool)
+            self.assertEqual('https://cwe.mitre.org/data/definitions/120.html', finding.references)
         with self.subTest(i=53):
             finding = findings[53]
             self.assertEqual("buffer/sscanf:The scanf() family's %s operation, without a limit specification, permits buffer overflows (CWE-120, CWE-20).", finding.title)
             self.assertEqual("Critical", finding.severity)
-            self.assertEqual("The scanf() family's %s operation, without a limit specification, permits buffer overflows (CWE-120, CWE-20).", finding.description)
+            description = '''**Result message:** buffer/sscanf:The scanf() family's %s operation, without a limit specification, permits buffer overflows (CWE-120, CWE-20).
+**Snippet:**       if (sscanf(argv[i], "%[^=]=%s", name, val) == 2) {
+**Rule name:** buffer/sscanf
+**Rule short description:** The scanf() family's %s operation, without a limit specification, permits buffer overflows (CWE-120, CWE-20).
+'''
+            self.assertEqual(description, finding.description)
             self.assertEqual("src/cli_main.cc", finding.file_path)
             self.assertEqual(482, finding.line)
             self.assertEqual("FF1021", finding.vuln_id_from_tool)
+            self.assertEqual('https://cwe.mitre.org/data/definitions/120.html', finding.references)
```

Comment:
better to use `self.assertIn('http://...', finding.references)`

Reply:
Why? This is exactly what I expect to be in the references.

Reply:
most of the parsers use this syntax for references:
```Markdown
* [URL TITLE](https:///example.com/1)
* [URL2 TITLE](https:///example.com/2)
```
I want to keep them aligned.

Reply:
I still don't understand to be honest. This is a test case, it just checks that it gets exactly the value it expects from the scan file.

Reply:
Ok just saw that we take the reference from rule data and we can't have more than one URL or a text instead. So we are not in the same case as the other parsers.